### PR TITLE
feat (networking): [dns resolution] cloud-init search backend domain using Azure provided DNS address

### PR DESCRIPTION
**Pull request by ferantivero**
- change netplan configuration using cloud-init: add per domain exception for backend to resolve using Azure provided DNS
- combine Netplan with special systemd-resolved config via cloud-init  to get global dns resolution for the rest of domains
- remove TODO that resolved DNS queries from hub

### CODE REVIEW

File: 07-compute-infra.md [hunk 1]
```diff
@@ -18,11 +18,16 @@ Azure Application Gateway, for this reference implementation, is placed in the s
 
 ## Steps
 
+1. Convert your frontend cloud-init (users) file to Base64.
```
Comment (ckittel) [inline 07-compute-infra.md:21]:
```suggestion
1. Convert your frontend cloud-init (network configuration) file to Base64.
```

File: 07-compute-infra.md [hunk 2]
```diff
@@ -18,11 +18,16 @@ Azure Application Gateway, for this reference implementation, is placed in the s
...
+
+   ```bash
+   FRONTEND_CLOUDINIT_BASE64=$(base64 workload-team/workload/frontendCloudInit.yml | tr -d '\n')
+   ```
```
Comment (ckittel) [inline 07-compute-infra.md:25]:
```suggestion
   ```

```

File: workload-team/app-infra-stamp.bicep [hunk 3]
```diff
@@ -38,6 +38,10 @@ param adminSecurityPrincipalObjectId string
 ])
 param adminSecurityPrincipalType string
 
+@description('A cloud init file (starting with #cloud-config) as a base 64 encoded string used to perform image customization on the jump box VMs. Used for user-management in this context.')
```
Comment (ckittel) [inline workload-team/app-infra-stamp.bicep:41]:
```suggestion
@description('A cloud init file (starting with #cloud-config) as a Base64 encoded string used to perform OS configuration on the VMs as part of bootstrapping. Used for network configuration in this context.')
```

File: workload-team/app-infra-stamp.bicep [hunk 4]
```diff
@@ -1030,21 +1036,6 @@ resource contosoPrivateDnsZone 'Microsoft.Network/privateDnsZones@2020-06-01' =
...
-  // TODO: This is going to need to be solved. As this is configured below, it will not work in this topology.
-  // THIS IS BEING DONE FOR SIMPLICTY IN DEPLOYMENT, NOT AS GUIDANCE.
-  // Normally a workload team wouldn't have this permission, and a DINE policy
-  // would have taken care of this step.
-  resource linkToHub 'virtualNetworkLinks' = {
-    name: 'to_${hubVirtualNetwork.name}'
-    location: 'global'
-    properties: {
-      virtualNetwork: {
-        id: hubVirtualNetwork.id
```
Comment (ckittel) [inline workload-team/app-infra-stamp.bicep:1043]: Shouldn't need the `existing` `hubVirtualNetwork` resource anymore at the top of the file.  Please remove.

File: workload-team/main.bicep [hunk 5]
```diff
@@ -68,6 +68,10 @@ param adminSecurityPrincipalObjectId string
 ])
 param adminSecurityPrincipalType string
 
+@description('A cloud init file (starting with #cloud-config) as a base 64 encoded string used to perform image customization on the jump box VMs. Used for user-management in this context.')
```
Comment (ckittel) [inline workload-team/main.bicep:71]:
```suggestion
@description('A cloud init file (starting with #cloud-config) as a Base64 encoded string used to perform OS configuration on the VMs as part of bootstrapping. Used for network configuration in this context.')
```

**Review by ckittel — CHANGES_REQUESTED** [review]

File: workload-team/app-infra-stamp.bicep (hunk 4 above)
Comment (ferantivero) [reply workload-team/app-infra-stamp.bicep:1043]: thanks for pointing this out @ckittel, 

done | addressed from 1d0042c8d2bd8cea8a6b45ce098bca51f12b50e8

**Review by ckittel — APPROVED** [review]

**Review by ckittel — APPROVED** [review]
LGTM (again).

You're welcome to merge, and then we can do any adjustements after meeting with Jeff.

File: 08-bootstrap-validation.md [hunk 6]
```diff
@@ -79,6 +79,36 @@ A web server is enabled on both tiers of this deployment so that you can test en
...
    > Ideally you'd just run `az network bastion ssh -n $AB_NAME_HUB -g rg-plz-connectivity-regional-hubs --target-resource-id <VM_RESOURCE_ID> --auth-type AAD` but due to a [known bug](https://github.com/Azure/azure-cli-extensions/issues/6408) you must connect using the above Azure Bastion tunnel method.
 
+   1. Validate DNS resolution from the frontend VM.
+
+      ```bash
+      sudo tcpdump -w /tmp/dns.pcap -i eth0 port 53 &
+      host azure.com && host backend-00.iaas-ingress.contoso.com
+      ```
+
+     > Note: the Frontend VMs are configured to route-only backend-00.iaas-ingress.contoso.com via Azure Provided DNS (168.63.129.16). The rest of the DNS queries should be resolved via hub DNS IP.
```
Comment (ckittel) [inline 08-bootstrap-validation.md:89]:
```suggestion
     > The frontend VM are configured to resolve backend-00.iaas-ingress.contoso.com via Azure DNS (168.63.129.16). All other DNS queries are resolved by the virtual network provided (DHCP) DNS IP, which is the Azure Firewall DNS proxy in the hub.
```

File: 08-bootstrap-validation.md [hunk 7]
```diff
@@ -79,6 +79,36 @@ A web server is enabled on both tiers of this deployment so that you can test en
...
+      <ctrl+c>
+      ```
+
+   1. Examine the captured packets
+
+      ```bash
+      tcpdump -tnr /tmp/dns.pcap | grep -E 'backend-00.iaas-ingress.contoso.com|azure.com'
+      ```
+
+      > Note: the results should reflect that your backend VM and azure FQDNs are resolved using diffrent DNS IPs.
```
Comment (ckittel) [inline 08-bootstrap-validation.md:104]:
```suggestion
      > The results should reflect that your backend FQDN and other FQDNs are resolved using different DNS IPs.
```

**Review by ckittel — APPROVED** [review]
Minor edits.  No need for me to approve again after you apply these.

**Pull request merged**